### PR TITLE
Implement the rejection feedback survey on candidate interface

### PR DESCRIPTION
**Pull request by JR-G**
## Context

We'd like to know whether candidates find the rejection reasons given by providers helpful or not, and which types of feedback are more helpful than others.

To do this, we’d like to trial a one-question survey on the candidate interface asking "Is this feedback helpful?" with Yes and No buttons.

## Changes proposed in this pull request

- Introduce a new service that will use the Reject Feedback model to associate the feedback response
- Add a new component that will render the feedback buttons
- Include this within the application dashboard

|Before|After|
|---|---|
|image|image|

|Feedback helpful|Feedback not helpful|
|---|---|
|image|image|

## Link to Trello card

https://trello.com/c/5FcCz7Q5/801-implement-is-this-feedback-helpful-survey-on-candidate-interface

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/frontend/styles/_feedback.scss [hunk 1]
```diff
@@ -8,3 +8,26 @@
   padding: govuk-spacing(2) govuk-spacing(5);
   min-height: 80px;
 }
+
+/* 'Is this feedback? helpful?' component
+This is shown on the application interface next to feedback
+received after an application has been rejected. */
```
[the remark this answers is not in the document]
Comment (JR-G) [reply app/frontend/styles/_feedback.scss:14]: 🪓

**Comment by carlosmartinez** [comment]
looks good. creating a review app and will have a go.

**Review by frankieroberto — APPROVED** [review]
Tested in the review app. seems to work!

File: app/components/candidate_interface/application_dashboard_course_choices_component.rb [hunk 2]
```diff
@@ -163,13 +163,16 @@ def rejection_reasons_row(application_choice)
       return unless application_choice.rejected?
       return unless application_choice.rejection_reason.present? || application_choice.structured_rejection_reasons.present?
 
+      activate_rejection_feedback_button = true if FeatureFlag.active?(:is_this_feedback_helpful_survey)
```
Comment (carlosmartinez) [inline app/components/candidate_interface/application_dashboard_course_choices_component.rb:166]:
```suggestion
      activate_rejection_feedback_button = FeatureFlag.active?(:is_this_feedback_helpful_survey)
```

File: app/controllers/candidate_interface/rejection_feedback_survey_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module CandidateInterface
+  class RejectionFeedbackSurveyController < CandidateInterfaceController
+    def new
+      ProvideRejectionFeedback.new(application_choice_params, helpful_params).call
+      redirect_to candidate_interface_application_complete_path
+      flash[:success] = 'Feedback successfully provided'
```
Comment (carlosmartinez) [inline app/controllers/candidate_interface/rejection_feedback_survey_controller.rb:6]: is the overloading of the word "feedback" a bit confusing here? we've asked "is this feedback helpful?" and now we're telling them "feedback successfully provided". perhaps something along the lines of "thanks, your response has been recorded".

Comment (frankieroberto) [reply app/controllers/candidate_interface/rejection_feedback_survey_controller.rb:6]: Hopefully we'll be able to remove this flash message altogether (for most users) by switching up to an AJAX submission in future...

File: spec/components/shared/rejection_reasons/rejection_feedback_survey_component_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe RejectionReasons::RejectionFeedbackSurveyComponent do
+  describe 'rendered component' do
+    let(:application_choice) { create(:application_choice, :with_rejection) }
+
+    context 'when no response has been provided' do
+      it 'renders the survey button' do
+        result = render_inline(described_class.new(application_choice:))
+        expect(result.text).to include('Is this feedback helpful?')
```
[the remark this answers is not in the document]
Comment (frankieroberto) [reply spec/components/shared/rejection_reasons/rejection_feedback_survey_component_spec.rb:10]: cue big debate about whether tests should look up interface text from the locale file implementation or not... 🔥

File: app/components/candidate_interface/application_dashboard_course_choices_component.rb (hunk 2 above)
Comment (JR-G) [reply app/components/candidate_interface/application_dashboard_course_choices_component.rb:166]: 🧹

File: spec/components/shared/rejection_reasons/rejection_feedback_survey_component_spec.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (JR-G) [reply spec/components/shared/rejection_reasons/rejection_feedback_survey_component_spec.rb]: have put the text into yml (for now...)

**Pull request merged**